internal/controller/controller.go

```diff
@@ -84,37 +89,37 @@ func (c *Controller) Start(ctx context.Context) error {
 	metadataCollector.AddCustomData("host.name", hostname)
 	metadataCollector.AddCustomData("host.ip", sourceIP)
 
-	// Network operations to CA start here
-	var rep reporter.Reporter
-	// Connect to the collection agent
-	rep, err = reporter.Start(ctx, &reporter.Config{
-		CollAgentAddr:          c.config.CollAgentAddr,
-		DisableTLS:             c.config.DisableTLS,
-		MaxRPCMsgSize:          32 << 20, // 32 MiB
-		MaxGRPCRetries:         5,
-		GRPCOperationTimeout:   intervals.GRPCOperationTimeout(),
-		GRPCStartupBackoffTime: intervals.GRPCStartupBackoffTime(),
-		GRPCConnectionTimeout:  intervals.GRPCConnectionTimeout(),
-		ReportInterval:         intervals.ReportInterval(),
-		CacheSize:              traceHandlerCacheSize,
-		SamplesPerSecond:       c.config.SamplesPerSecond,
-		KernelVersion:          kernelVersion,
-		HostName:               hostname,
-		IPAddress:              sourceIP,
-	})
-	if err != nil {
-		return fmt.Errorf("failed to start reporting: %w", err)
+	if c.reporter == nil {
```

Comment:
Start() should only be called once,  maybe put that into the function comment.
So no need to check `c.reporter` here.

Reply:
We do need to check `c.reporter`.
If we have a custom one, we'd already have one and we don't want to overwrite it.

Reply:
Maybe I miss the point. Where exactly is `c.reporter` set if not in `Start`?

Reply:
With the `WithReporter` option when initializing the controller.
`reporter.Start` always initializes an OTLP reporter. We need a "custom" reporter to use the one in #208.

```golang
controller.New(&controller.Config{}, controller.WithReporter(reporter.NewCollectorReporter()))
```

Reply:
And since a reporter is mandatory to the controller, we can drop the `WithReporter` function.

Reply:
I can do that, but it's not just about passing a reporter.
Outside of the breaking change, the current OTLP reporter is very stateful (it relies on a package method, sets system-wide limits), while I'm building a stateless reporter.
And nothing has tests. So it feels very prone to mistakes.

Reply:
I've started doing that in c9f21866155804099caaf9f5cb66e7058539c7ed
For this to be OK, the PR will become rather bigger though, because some of the OTLP config is retrieved within the collector and either needs to be moved away from it, or duplicated.